python/ray/experimental/compiled_dag_ref.py

```diff
@@ -0,0 +1,37 @@
+import ray
+
+
+class CompiledDAGRef(ray.ObjectRef):
+    def __init__(
+        self,
+        dag: "ray.experimental.CompiledDAG",
+        execution_index: int,
+    ):
+        self._dag = dag
+        self._execution_index = execution_index
+        self._called = False
+        self._reader_refs = (
+            dag.dag_output_channels._reader_ref
+            if dag.has_single_output
+            else [channel._reader_ref for channel in dag.dag_output_channels]
+        )
+
+    def __repr__(self):
+        return (
+            f"CompiledDAGRef(_dag={self._dag}, "
+            f"_execution_index={self._execution_index}, "
+            f"_called={self._called}, "
+            f"_reader_refs={self._reader_refs})"
+        )
+
+    def __copy__(self):
+        raise ValueError("CompiledDAGRef cannot be copied.")
+
+    def __deepcopy__(self, memo):
+        raise ValueError("CompiledDAGRef cannot be deep copied.")
+
+    def get(self):
+        if self._called:
+            raise ValueError("ray.get() was already called on this CompiledDAGRef.")
```

Comment:
Why do we not allow this?

Reply:
Yes, let's make the error a bit better. It'd be good to including something about how ray.get() can only be called once on CompiledDAGRefs.

Reply:
Updated. Also added class doc.